internal/k8s/controller_test.go

```diff
@@ -229,108 +230,94 @@ func TestHasCorrectIngressClass(t *testing.T) {
 	}
 }
 
-func TestHasCorrectIngressClassVS(t *testing.T) {
-	ingressClass := "ing-ctrl"
-	lbcIngOnlyTrue := &LoadBalancerController{
-		ingressClass:        ingressClass,
+func deepCopyWithIngressClass(obj interface{}, class string) interface{} {
```

Comment:
Not sure about this function, all the paths looks the same.
Can it be replaced on line 315 by 
```
                objCopy := obj.DeepCopy()
 		objCopy.Spec.IngressClass = class
```

Reply:
@soneillf5 
we need to reference a concrete type to apply `objCopy.Spec.IngressClass = class`  , that's why we have a a type switch with 3 cases

Reply:
if deepCopy is confusing, I can create a concrete type instead, but still do that three times:
```
	case *conf_v1.VirtualServer:
		vs := conf_v1.VirtualServer{}
		vs.Spec.IngressClass = class
		return &vs
```

Reply:
No that's fine as it is, thanks for the explanation.